[Aisyah Ally] iP

aisyahally opened the pull request:
### Simba

> "Memory is the diary that we all carry about with us." - Oscar Wilde

Simba frees your mind of having to remember things you need to do. It's

- text-based 💬💬
- easy to learn 📚📚
- fast to use 💨💨

All you need to do is

1. download it from [here](https://github.com/nus-cs2103-AY2425S2/ip/pull/426)
2. add your tasks
3. let it manage your tasks for you

And it is **FREE**!

Features:

- [X] managing tasks
- [X] managing deadlines
- [ ] reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public static void main(String[] args) {
        new Simba("simba.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Level-0

Commit:
Revert "Level-0"

This reverts commit e9cc156c934738f2f1d5c1934f3e55e0955358d2.

Commit:
add greeting

Commit:
add echo

Commit:
add list

Commit:
add mark as done

Commit:
add todo, event, deadline

Commit:
add automated text ui testing

Commit:
add error handling

Commit:
add deleting

Commit:
Modify greeting

Commit:
Save tasks in a separate file

Commit:
Revert "Save tasks in a separate file"

This reverts commit 94e9f94803ef3bd551348e442ad06d6bd6c3cfe1.

Commit:
Reapply "Save tasks in a separate file"

This reverts commit 0f1bc9b71e02e55382cdfc402195bdd75a28e3d9.

Commit:
Import LocalDateTime

Commit:
Add date and time

Commit:
Separate into different classes

Commit:
Put Simba into a package

Commit:
Merge commit '81a9c537afe8492675eb253968939364d65db091'

Commit:
Add JUnit tests

Commit:
Package app as a JAR file

Commit:
Add JavaDoc comments

Commit:
Add finding task by keyword

Commit:
Tweak to coding standard

Commit:
Rearrange import statements

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'branch-A-Level-9'

Commit:
Parse from first word only

nicholasohjj commented on src/test/java/simba/ui/SimbaTest.java:
```diff
@@ -0,0 +1,22 @@
+package simba.ui;  //same package as the class being tested
```
Suggestion:
Move the comment to a separate line and ensure there is a blank line after the package declaration.

nicholasohjj replied:
```suggestion
// This test class belongs to the same package as the class being tested.
package simba.ui;
```

Suggestion:
Move the comment to a separate line and ensure there is a blank line after the package declaration.

nicholasohjj commented on src/test/java/simba/ui/SimbaTest.java:
```diff
@@ -0,0 +1,22 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.ArrayList;
```
The import statement `import java.util.ArrayList;` is **unused** in this test file.
```suggestion
```
Remove the unused import to keep the code clean.

nicholasohjj commented on src/test/java/simba/ui/SimbaTest.java:
```diff
@@ -0,0 +1,22 @@
+import java.util.ArrayList;
+
+public class SimbaTest {
+    Parser parser1 = new Parser("delete 1");
+    Parser parser2 = new Parser("delete 2");
```
`parser1` and `parser2` are **declared at the class level**, making them shared across multiple test methods.

It is generally cleaner to declare variables within each test method to ensure test independence.

nicholasohjj commented on src/test/java/simba/ui/SimbaTest.java:
```diff
@@ -0,0 +1,22 @@
+    public void dummyTest(){
+        assertEquals(1, parser1.idxToDelete());
+    }
+
```
The test method names `dummyTest` and `anotherDummyTest` do not describe the actual behaviour being tested.

Use descriptive test method names that clarify what each test checks.

nicholasohjj reviewed (changes requested):
Overall, your code follows a clear structure and is easy to understand. However, I noticed a few coding standard violations that could be improved for better readability and maintainability.

Commit:
Separate into files

Commit:
Add checkstyle

Commit:
Add new line

Commit:
Merge branch 'branch-A-Checkstyle'

Commit:
Start adding GUI

Commit:
Made small changes

Ryuse commented on src/main/java/simba/ui/NonsenseException.java:
```diff
@@ -0,0 +1,7 @@
+/**
+ * Thrown when an invalid command is entered.
+ */
+public class NonsenseException extends Exception {
```
Maybe `InvalidCommandException` might be a better name?

Ryuse commented on src/main/java/simba/ui/Parser.java:
```diff
@@ -0,0 +1,110 @@
+     * @return The task to add.
+     * @throws Exception If the command is invalid or the task description is empty.
+     */
+    Task taskToAdd() throws Exception {
```
There seems to be quite a bit of nesting in this function. 

Might it be better to modularize parts of the function into other functions for better readability and prevention of deep nesting?

Ryuse commented on src/main/java/simba/ui/Simba.java:
```diff
@@ -0,0 +1,65 @@
+     * Runs the Simba application, displaying a welcome message and processing user commands.
+     */
+    public void run() {
+        System.out.println("\tHello I am Simba :D");
```
Maybe all the print statements can be put all into a function and be put in `Ui.java?`

Ryuse commented on src/main/java/simba/ui/Ui.java:
```diff
@@ -0,0 +1,89 @@
+     *
+     * @param command The user command to process.
+     */
+    void readCommand(String command) {
```
Might it be better to put the print statements into functions depending on their commands?

Ryuse reviewed (commented):
Well documented code. Maybe putting parts of the code into separate functions might make it better?

Commit:
Add GUI until part 3

Commit:
Use FXML

Commit:
Change code to return strings instead of printing

Commit:
Change NonsenseException to InvalidCommandException

Commit:
Merge branch 'branch-Level-10'

Commit:
Rename functions

Seethevon commented on src/main/java/simba/ui/DialogBox.java:
```diff
@@ -0,0 +1,47 @@
+package simba.ui;
```
I like the clarity and consistency in naming as it makes the code more readable and neat. Good job 👍 .

Seethevon commented on src/main/java/simba/ui/Main.java:
```diff
@@ -2,23 +2,91 @@
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+    private Image simbaImage = new Image(this.getClass().getResourceAsStream("/images/simba.png"));
```
Do the values of simbaImage and userImage change? If it doesn't then could it be a constant? 🤔

Commit:
Add assert to readCommand of Ui class

Command should always be a valid string. Assert will exit the function if command is null or empty.

Commit:
Edit JavaDoc comments

JavaDoc comments are updated after functions were changed.

Commit:
Edit JavaDoc comments

Functions were edited to follow code quality, increasing readability by breaking the if-else statements into smaller functions. JavaDoc comments are updated.

Commit:
Merge branch 'branch-A-Assertions'

Branch 'branch-A-Assertions' merged to master.

Commit:
Merge branch 'branch-A-CodeQuality'

Branch 'branch-A-CodeQuality' merged to master.

Commit:
Add more JavaDoc comments

Some new functions were missing JavaDoc comments.

Commit:
Add sorting function for tasks

Automatically sorts tasks when they are added to the task list. Tasks are sorted first by type, date (if applicable), then name.

Commit:
Edit GUI

Colours and fonts were changed for better readability and aesthetic reasons.

Commit:
Add More Error Handling

Error handling for invalid dates for events added. Easier management of tasks with constant indexing.

Commit:
Add screenshot and user guide

Commit:
Edit user guide

Commit:
Edit image in user guide

Commit:
Edit image

Commit:
Edit user guide

Commit:
Edit user guide

Commit:
Close window with bye

Commit:
Add JavaDoc comments

JavaDoc comments added for all methods in all classes.

aisyahally committed:
Merge pull request #2 from aisyahally/branch-javadoc

Add JavaDoc comments

Commit:
Change tested method name in JUnit test

Method was renamed in previous steps.
The new name is updated here.

Commit:
Rename package name to create jar file

Commit:
Edit storage to retain list between runs

Storage previously wrote a new file each time.
